src/test/java/org/fogbowcloud/manager/requests/api/local/http/AttachmentOrdersControllerTest.java

```diff
@@ -0,0 +1,395 @@
+
+        // exercise: Make the request
+        result = this.mockMvc.perform(requestBuilder).andReturn();
+
+        // verify
+        expectedStatus = HttpStatus.UNSUPPORTED_MEDIA_TYPE.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        Mockito.verify(this.facade, Mockito.times(1))
+    		.createAttachment(Mockito.any(AttachmentOrder.class), Mockito.anyString());
+
+        // Case 3: Json with empty properties.
+        // set up
+        requestBuilder =
+                createRequestBuilder(
+                        HttpMethod.POST,
+                        ATTACHMENT_ENDPOINT,
+                        getHttpHeaders(),
+                        BODY_WITH_EMPTY_PROPERTIES);
+
+        // exercise: Make the request
+        result = this.mockMvc.perform(requestBuilder).andReturn();
+
+        // verify
+        expectedStatus = HttpStatus.UNSUPPORTED_MEDIA_TYPE.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        Mockito.verify(this.facade, Mockito.atLeast(2))  // for the two tests where a non-empty body was provided
+    		.createAttachment(Mockito.any(AttachmentOrder.class), Mockito.anyString());
+    }
+
+    // test case: Request the list of all attachments when the facade returns an empty list. 
+    // Check the response of request and the call of facade for get the attachments. 
+    @Test
+    public void getAllAttachmentsWhenHasNoData() throws Exception {
+    	//set up
+        List<AttachmentInstance> attachementInstanceList = new ArrayList<>();
+        Mockito.doReturn(attachementInstanceList).when(this.facade).getAllAttachments(Mockito.anyString());
+
+        RequestBuilder requestBuilder =
+                createRequestBuilder(HttpMethod.GET, ATTACHMENT_ENDPOINT, getHttpHeaders(), "");
+
+        // exercise: Make the request
+        MvcResult result = this.mockMvc.perform(requestBuilder).andReturn();
+
+        // verify
+        int expectedStatus = HttpStatus.OK.value();
+        Assert.assertEquals(expectedStatus, result.getResponse().getStatus());
+        
+        String expectedResult = "[]";
+
+        Assert.assertEquals(expectedResult, result.getResponse().getContentAsString());
+        
+        Mockito.verify(this.facade, Mockito.times(1))
+			.getAllAttachments(Mockito.anyString());
+    }
+
+    // test case: Request the list of all attachments when the facade returns an non-empty list. 
+    // Check the response of request and the call of facade for get the attachments. 
+    @Test
+    public void getAllAttachmentsWhenHasData() throws Exception {
+    	// set up
+        AttachmentInstance AttachmentInstance1 = new AttachmentInstance("fake-Id-1");
+        AttachmentInstance AttachmentInstance2 = new AttachmentInstance("fake-Id-2");
+        AttachmentInstance AttachmentInstance3 = new AttachmentInstance("fake-Id-3");
+
+        List<AttachmentInstance> AttachmentInstanceList =
+                Arrays.asList(
```

Comment:
the Arrays.asList method is variadic. There is no need for the array declaration